Frameworks/Foundation/NSException.mm

```diff
@@ -216,7 +244,18 @@ - (HRESULT)_hresult {
         return [hresultValue unsignedIntValue];
     }
 
-    TraceWarning(L"NSException", L"Called -hresult on an NSException without an HRESULT!");
-    return E_UNEXPECTED;
+    return [[self class] _HRESULTForExceptionName:self.name];
+}
+
+- (void)_processException {
+    HRESULT hr = [self _hresult];
+    HSTRING_HEADER header;
+    HSTRING hsReason = nullptr;
+    
+    if(self.reason != nil) {
+        LPCWSTR reason = (LPCWSTR)([[self reason] cStringUsingEncoding:NSUnicodeStringEncoding]);
+        WindowsCreateStringReference(reason, wcslen(reason), &header, &hsReason);
+    }
+    RoOriginateError(hr, hsReason);
```

Comment:
How is that "weird"?  It is clearly documented.


---
In reply to: [93300590](https://github.com/Microsoft/WinObjC/pull/1570#discussion_r93300590) [](ancestors = 93300590,93300439,93125265)